common4j/src/main/com/microsoft/identity/common/java/authorities/AuthorityDeserializer.java

```diff
@@ -60,12 +61,15 @@ public Authority deserialize(JsonElement json, Type typeOfT, JsonDeserialization
 
                     if (aadAuthority != null) {
                         try {
-                            final CommonURIBuilder uri = new CommonURIBuilder(aadAuthority.getAuthorityUri());
+                            // Use authority_url provided by the developer, otherwise, use the audience cloudUrl from aadAuthority.getAuthorityUri()
```

Comment:
The fix should be in AzureActiveDirectoryAuthority.

Reply:
> My understanding is that this is a regression.
> 
> If that's the case, could you please go back to the non-regressed one and see what was the previous behavior?
> 
> If the app sets BOTH authority and authority_url, which one would have a higher precedence?

Yes, I based my implementation on the non-regressed one. Here's a snipped of what we had before in the `AuthorityDeserializer`

```java
                    // The developer might supply authority_url instead of audience.
                    // In that case, we'll try our best to map the audience here.
                    if (aadAuthority != null && aadAuthority.mAuthorityUrl != null) {
                        final Uri uri = Uri.parse(aadAuthority.mAuthorityUrl);
                        final String cloudUrl = uri.getScheme() + "://" + uri.getHost();
                        final String tenant = uri.getLastPathSegment();
                        if (!TextUtils.isEmpty(tenant)) {
                            aadAuthority.mAudience = AzureActiveDirectoryAudience.getAzureActiveDirectoryAudience(cloudUrl, tenant);
                        }
                    }
```